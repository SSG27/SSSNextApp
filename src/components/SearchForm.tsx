import React from 'react';
import '@/styles/SearchForm.css'

interface SearchFormProps {
  handleSubmit: (event: React.FormEvent) => void;
  handleChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  inputs: {
    movietitle: string;
    moviecountry: string;
  };
}

const SearchForm: React.FC<SearchFormProps> = ({ handleSubmit, handleChange, inputs }) => {
  return (
    <form onSubmit={handleSubmit}>
      <br></br>
      <label>Enter the movie title: 
        <input type="text" 
        name = "movietitle"
        value={inputs.movietitle || ""} 
        onChange={handleChange}
        />
      </label>
      <br></br>
      <label>Enter the two letter country code (e.g gb): 
        <input type="text"name = "moviecountry"
        value={inputs.moviecountry || ""} 
        onChange={handleChange}
        />
      </label>
      <br></br>
      {/* creates submit button */}
      <input type="submit" className='button'/>
    </form>
  );
};

export default SearchForm;
