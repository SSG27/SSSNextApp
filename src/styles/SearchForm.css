.button {
    background-color: #426599;
    color: #FFFFFF;
    border-radius: 8px;
    border: none;
}