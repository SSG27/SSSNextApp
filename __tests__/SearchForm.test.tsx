import React from 'react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import SearchForm from '@/components/SearchForm';

it('updates state on user input', () => {
  // Create a snapshot for the initial rendering
  const component = renderer.create(
    <SearchForm 
      handleSubmit={() => {}} 
      handleChange={() => {}} 
      inputs={{ movietitle: '', moviecountry: '' }} 
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // Manually trigger user input for movie title
  renderer.act(() => {
    userEvent.type(tree.children[1].children[1], 'dunkirk'); // Assuming the movie title input is at index 1
  });

  // Re-rendering after user input
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // Manually trigger user input for country code
  renderer.act(() => {
    userEvent.type(tree.children[3].children[1], 'gb'); // Assuming the country code input is at index 3
    userEvent.click(tree.children[5]);
  });

  // Re-rendering after user input
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
