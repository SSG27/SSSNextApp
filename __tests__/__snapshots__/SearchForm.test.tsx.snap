// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updates state on user input 1`] = `
<form
  onSubmit={[Function]}
>
  <br />
  <label
    className="form"
  >
    Enter the movie title:
    <input
      name="movietitle"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <br />
  <label>
    Enter the two letter country code (e.g gb):
    <input
      name="moviecountry"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <br />
  <input
    className="button"
    data-testid="submit-button"
    type="submit"
  />
</form>
`;

exports[`updates state on user input 2`] = `
<form
  onSubmit={[Function]}
>
  <br />
  <label
    className="form"
  >
    Enter the movie title:
    <input
      name="movietitle"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <br />
  <label>
    Enter the two letter country code (e.g gb):
    <input
      name="moviecountry"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <br />
  <input
    className="button"
    data-testid="submit-button"
    type="submit"
  />
</form>
`;

exports[`updates state on user input 3`] = `
<form
  onSubmit={[Function]}
>
  <br />
  <label
    className="form"
  >
    Enter the movie title:
    <input
      name="movietitle"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <br />
  <label>
    Enter the two letter country code (e.g gb):
    <input
      name="moviecountry"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <br />
  <input
    className="button"
    data-testid="submit-button"
    type="submit"
  />
</form>
`;
